SpatialGDK/Source/SpatialGDKServices/Private/LocalReceptionistProxyServerManager.cpp

```diff
@@ -0,0 +1,265 @@
+	if (bSuccess)
+	{
+		if (State.Contains("LISTENING"))
+		{
+			if (GetProcessName(OutPID, ProcessName))
+			{
+				OutLogMsg = TEXT("%s process with PID: %s"), ProcessName, OutPID;
+				return true;
+			}
+
+			OutLogMsg = TEXT("Unknown process with PID: %s."), *OutPID;
+
+			return true;
+		}
+	}
+	OutLogMsg = TEXT("No Process is blocking the required port.");
+
+	return false;
+}
+
+bool FLocalReceptionistProxyServerManager::LocalReceptionistProxyServerPreRunChecks(int32 ReceptionistPort)
+{
+	FString OutLogMessage;
+	FString PID;
+
+	// Check if any process is blocking the receptionist port
+	if (CheckIfPortIsBound(ReceptionistPort, PID, OutLogMessage))
+	{
+		// Try killing the process that blocks the receptionist port if the process blocking the port is a previously running proxy.
+		if (PID == ParsePid())
+		{
+			bool bProcessKilled = SpatialCommandUtils::TryKillProcessWithPID(PID);
+			if (!bProcessKilled)
+			{
+				UE_LOG(LogLocalReceptionistProxyServerManager, Warning, TEXT("Failed to kill the process that is blocking the port. %s"), *OutLogMessage);
+				return false;
+			}
+
+			UE_LOG(LogLocalReceptionistProxyServerManager, Log, TEXT("Succesfully killed %s"), *OutLogMessage);
+			return true;
+		}
+
+		UE_LOG(LogLocalReceptionistProxyServerManager, Log, TEXT("The required port is blocked from a different process with Pid: %s"), *PID);
+		return false;
+	}
+
+	UE_LOG(LogLocalReceptionistProxyServerManager, Log, TEXT("The required port is not blocked!"));
+	return true;
+}
+
+void FLocalReceptionistProxyServerManager::Init(int32 Port)
+{
+	if (!IsRunningCommandlet())
+	{
+		LocalReceptionistProxyServerPreRunChecks(Port);
+	}
+}
+
+
+bool FLocalReceptionistProxyServerManager::TryStopReceptionistProxyServer()
+{
+	if (ProxyServerProcHandle.IsValid())
+	{
+		SpatialCommandUtils::StopLocalReceptionistProxyServer(ProxyServerProcHandle);
+		bProxyIsRunning = false;
+		return true;
+	}
+
+	return false;
+}
+
+
+TSharedPtr<FJsonObject> FLocalReceptionistProxyServerManager::ParsePidFile()
+{
+	FString ProxyInfoFileResult;
+	TSharedPtr<FJsonObject> JsonParsedProxyInfoFile;
+
+	if (FFileHelper::LoadFileToString(ProxyInfoFileResult, *ProxyInfoFilePath))
+	{
+		if (FSpatialGDKServicesModule::ParseJson(ProxyInfoFileResult, JsonParsedProxyInfoFile))
+		{
+			return JsonParsedProxyInfoFile;
+		}
+		else
+		{
+			UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Json parsing of proxyInfo.json failed. Can't get proxy's Pid."));
+		}
+	}
+	else
+	{
+		UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Loading proxyInfo.json failed. Can't get proxy's Pid."));
+	}
+
+	return nullptr;
+}
+
+void FLocalReceptionistProxyServerManager::SetPidInJson(const FString& Pid)
+{
+	FString ProxyInfoFileResult;
+
+	// If file does not exist create an empty json file 
+	if (!FPaths::FileExists(ProxyInfoFilePath))
+	{
+		FFileHelper::SaveStringToFile(TEXT("{ }"), *ProxyInfoFilePath);
```

Comment:
How about declaring `JsonParsedProxyInfoFile` above this `if` as
```
TSharedPtr<FJsonObject> JsonParsedProxyInfoFile;
```
and just creating the empty json object here instead of saving a temporary file?
```suggestion
		JsonParsedProxyInfoFile = MakeShareable(new FJsonObject());
```

Reply:
That's creating an empty file when the file does not exist. So we can then later add the PID field in the file. If I do not create it here then the rest won't work.